### PR TITLE
Fix CourseRunStatus for course runs with fuzzy start date

### LINKED ISSUE #3737: Don't throw an error when courses don't have start dates
Sometimes we know that a course will be run before we know the exact start date. To handle these situations, we have the "fuzzy" start date. 

When there isn't a start date set, we should avoid raising an exception, and the user's dashboard should still render. 

ImproperlyConfigured: The course course-v1:MITx+CTL.CFx+1T18U1 results are not either current, past, or future at the s...

https://sentry.io/mit-office-of-digital-learning/micromasters/issues/425753323/

```
ImproperlyConfigured: The course course-v1:MITx+CTL.CFx+1T18U1 results are not either current, past, or future at the same time
(2 additional frame(s) were not displayed)
...
  File "dashboard/api.py", line 177, in get_user_program_info
    response_data['programs'].append(get_info_for_program(mmtrack_info))
  File "dashboard/api.py", line 206, in get_info_for_program
    get_info_for_course(course, mmtrack)
  File "dashboard/api.py", line 260, in get_info_for_course
    for course_run in course.courserun_set.all()]
  File "dashboard/api.py", line 260, in <listcomp>
    for course_run in course.courserun_set.all()]
  File "dashboard/api.py", line 364, in get_status_for_courserun
    course_run.edx_course_key

ImproperlyConfigured: The course course-v1:MITx+CTL.CFx+1T18U1 results are not either current, past, or future at the same time
```

We should address #3720 at the same time.

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #3737

#### What's this PR do?
This allows a learner to access the dashboard when enrolled in a course run without a start_date.

#### How should this be manually tested?
Enroll a learner in some course run, then delete the start date and set the fuzzy start date for that course run. You should be able to load the dashboard without errors.

### CODE REVIEW

**Comment by pdpinch** [comment]
Does this also fix #3720 ?

**Comment by annagav** [comment]
No, it doesn't. Those issues are not related to the same enrollment configuration.

File: courses/models.py [hunk 1]
```diff
@@ -218,6 +218,13 @@ def is_future(self):
             return False
         return self.start_date > now_in_utc()
 
+    @property
+    def is_promised(self):
+        """Checks if the course has fuzzy start date"""
```
Comment (mbertrand) [inline courses/models.py]: Add a `Returns:` documentation block here

Comment (mbertrand) [reply courses/models.py]: PS Just noticed the other PR for the `Course starts...` text

**Comment by annagav** [comment]
@mbertrand, I addressed the comment. Thanks!

**Pull request merged**